File: keycloak/iam_spi/src/main/java/de/intevation/iam/model/Institution.java
/* Copyright (C) 2022 by Bundesamt fuer Strahlenschutz
 * Software engineering by Intevation GmbH
 *
 * This file is Free Software under the GNU GPL (v>=3)
 * and comes with ABSOLUTELY NO WARRANTY!
 */
package de.intevation.iam.model;

import java.util.List;
import java.util.Map;

import org.keycloak.models.GroupModel;

public class Institution {
    private String id;

    private String name;

    private Map<String, List<String>> attributes;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Map<String, List<String>> getAttributes() {
        return attributes;
    }

    public void setAttributes(Map<String, List<String>> attributes) {
        this.attributes = attributes;
    }

    /**
     * Create an Institution from a GroupModel.
     * @param group GroupModel to convert
     * @return Institution instance
     */
    public static Institution fromGroupModel(GroupModel group) {
        Institution institution = new Institution();
        institution.setId(group.getId());
        institution.setName(group.getName());
        institution.setAttributes(group.getAttributes());
        return institution;
    }
}

File: client/src/store/institution.js
/* Copyright (C) 2022 by Bundesamt fuer Strahlenschutz
 * Software engineering by Intevation GmbH
 *
 * This file is Free Software under the GNU GPL (v>=3)
 * and comes with ABSOLUTELY NO WARRANTY!
 */
import { Promise } from "core-js";
import { HTTP } from "../lib/http";
export const institution = {
  namespaced: true,
  state: () => ({
    //List of institutions
    institutions: [],
    institutionNames: [],
    //Institution currently edited
    institution: {},
    //Institution to be created
    newInstitution: {
      name: "",
    },
    foundInstitutions: [],
  }),
  mutations: {
    //Convert current institution attributes to string arrays
    convertCurrentInstitutionAttributes: (state) => {
      let attributes = state.institution.attributes;
      for (let attribute in attributes) {
        if (typeof attributes[attribute] == "string") {
          attributes[attribute] = attributes[attribute].split(", ");
        }
      }
    },
    setInstitutionList: (state, data) => {
      state.institutions = data;
    },
    setFoundInstitutions: (state, data) => {
      state.foundInstitutions = data;
    },
    setInstitutionNames: (state, data) => {
      var names = [];
      data.forEach((institution) => {
        names.push(institution.name);
      });
      state.institutionNames = names;
    },
    setInstitution: (state, data) => {
      state.institution = data;
    },
    updateInstitutionEntity: (state, data) => {
      state.institutions.forEach((element, index) => {
        if (element.id === data.id) {
          state.institutions[index] = data;
        }
      });
    },
  },
  actions: {
    loadInstitutions({ commit }, searchString) {
      return new Promise((resolve, reject) => {
        HTTP.get("/institution", {
          params: { search: searchString },
        })
          .then((response) => {
            commit("setInstitutionList", response.data);
            resolve(response);
          })
          .catch((error) => reject(error));
      });
    },
    updateInstitution({ commit }, institution) {
      return new Promise((resolve, reject) => {
        HTTP.put("/institution", institution)
          .then((response) => {
            commit("updateInstitutionEntity", response.data);
            resolve(response);
          })
          .catch((error) => reject(error));
      });
    },
  },
};
